src/common/statistics/StatisticsManager.ts

```diff
@@ -0,0 +1,394 @@
+      stakeManagerEvents: this.allStakeManagerEvents,
+      relayHubConstructorParams,
+      deploymentBalances,
+      chainId,
+      runtimeVersion,
+      deploymentVersions,
+      contractsDeployment,
+      blockNumber,
+      totalStakesByRelays,
+      paymasters,
+      senders,
+      recipients,
+      relayServers,
+      totalGasPaidViaGSN
+    }
+  }
+
+  async fetchRelayHubEvents (): Promise<void> {
+    const transactionDepositedEventsData =
+      await this.contractInteractor.getPastEventsForHub([], { fromBlock: 1 }, [Deposited])
+    const depositedEvents = this.extractTransactionInfos<DepositedEventInfo>(transactionDepositedEventsData, Deposited)
+
+    const relayRegisteredEventsData =
+      await this.contractInteractor.getPastEventsForHub([], { fromBlock: 1 }, [RelayServerRegistered])
+    const relayRegisteredEvents = this.extractTransactionInfos<RelayRegisteredEventInfo>(relayRegisteredEventsData, RelayServerRegistered)
+
+    const transactionRelayedEventsData =
+      await this.contractInteractor.getPastEventsForHub([], { fromBlock: 1 }, [TransactionRelayed])
+    const transactionRelayedEvents = this.extractTransactionInfos<TransactionRelayedEventInfo>(transactionRelayedEventsData, TransactionRelayed)
+
+    const transactionRejectedEventsData =
+      await this.contractInteractor.getPastEventsForHub([], { fromBlock: 1 }, [TransactionRejectedByPaymaster])
+    const transactionRejectedEvents = this.extractTransactionInfos<TransactionRejectedByPaymasterEventInfo>(transactionRejectedEventsData, TransactionRejectedByPaymaster)
+
+    this.allRelayHubEvents = {
+      depositedEvents,
+      relayRegisteredEvents,
+      transactionRelayedEvents,
+      transactionRejectedEvents
+    }
+  }
+
+  // TODO
+  async getSendersInfo (): Promise<SenderInfo[]> {
+    return []
+  }
+
+  // TODO
+  async getRecipientsInfo (): Promise<RecipientInfo[]> {
+    return []
+  }
+
+  async getPaymastersInfo (): Promise<PaymasterInfo[]> {
+    const paymasters = new Set<Address>()
+    for (const depositedEventInfo of this.allRelayHubEvents.depositedEvents ?? []) {
+      paymasters.add(depositedEventInfo.returnValues.paymaster)
+    }
+    const paymasterInfos: PaymasterInfo[] = []
+    for (const address of paymasters) {
+      const relayHubBalance = (await this.contractInteractor.hubBalanceOf(address)).toString()
+      const acceptedTransactionsCount = this.allRelayHubEvents.transactionRelayedEvents.filter(it => isSameAddress(it.returnValues.paymaster, address)).length
+      const rejectedTransactionsCount = this.allRelayHubEvents.transactionRejectedEvents.filter(it => isSameAddress(it.returnValues.paymaster, address)).length
+      paymasterInfos.push({
+        address,
+        relayHubBalance,
+        acceptedTransactionsCount,
+        rejectedTransactionsCount
+      })
+    }
+    return paymasterInfos
+  }
+
+  extractUnique (events: Array<EventTransactionInfo<StakeChangeEvent>>): Address[] {
```

Comment:
extractUniqueRelayers

Reply:
and please, return a Set, not array.
the "`.includes()`" you use is O(n), and `set.has()` is O(1), and you do it in a loop.